platform/darwin/app/wideline.metal

```diff
@@ -0,0 +1,495 @@
+#include <metal_stdlib>
+#include "wideline.metal.h"
+
+using namespace metal;
+using namespace WhirlyKitShader;
+
+// Calculate fade based on time
+// (or) based on height.
+float calculateFade(constant Uniforms &uni,
+                    constant UniformDrawStateA &uniA)
+{
+    // Note: Feb 28 Meeting /w Steve
+    return 1;
+}
+
+// Get zoom from appropriate slot
+float ZoomFromSlot(constant Uniforms &uniforms,int zoomSlot)
+{
+    // Note: Feb 28 Meeting /w Steve
+    return 10.0;
+}
+
+struct IntersectInfo {
+    bool valid;
+    float2 interPt;
+    float c;
+    float ta,tb;
+};
+
+// wedge product (2D cross product)
+// if A ^ B > 0, A is to the right of B
+float wedge(float2 a, float2 b) {
+    return a.x * b.y - a.y * b.x;
+}
+
+// Intersect two lines
+IntersectInfo intersectLines(float2 a0, float2 a1, float2 b0, float2 b1)
+{
+    const float2 dA = a0 - a1;
+    const float2 dB = b0 - b1;
+
+    // Solve the system of equations formed by equating the lines to get their common point.
+    const float denom = wedge(dA, dB);
+    if (denom == 0.0) {
+        // If the denominator comes out zero, the lines are parallel and do not intersect
+        return { .valid = false };
+    }
+
+    const float tA = (a0.x * a1.y - a0.y * a1.x);
```

Comment:
Mostly the same shader source as before here? Refer to #2183 for previous nits I had in this shader.